### ISSUE TITLE
Tesla Power Wall (Solar Energy Storage) ☀️>🔋>💡 💻 ... 💰❓

### ISSUE THREAD

**Issue by nelsonic** [dwyl/home#25]
One of our Key goals **`@home`** is to be **100% Carbon Neutral** and **Energy Independent**<sup>1</sup>. 🎉 
We will partly achieve this goal by having an array of Solar Panels #20 on the roof. 🏡 ☀️ 
This is fine during the day (_and great on sunny days!_) but as we all (_hopefully_) know useless at night! 🌃 

If you're unfamiliar with Powerwall, watch: Fully Charged Tesla Powerwall 2: [youtu.be/nWLzlrGGuxQ](https://youtu.be/nWLzlrGGuxQ?t=75)
The product page is quite informative and they have their own videos: [tesla.com/en_GB/powerwall](https://www.tesla.com/en_GB/powerwall)
A relevant/unbiased article on battery storage and Powerwall economics:
https://www.energysage.com/solar/solar-energy-storage/tesla-powerwall-home-battery/

> While living in rainy London, I read _many_ articles (_both "media" and academic papers_) and watched _all_ the videos I could on energy storage<sup>2</sup> and concluded that at present the best price per kWh stored and _value-for-money_ system (_including the good App/interface_) is Tesla's system! According to _many_ reviews online the system has an _excellent_ battery management system (BMS) which means that capacity does not decline significantly with age, so the "useful life" of **15-years**

I have estimated that we will need **Two** Powerwall units for a total of **24kWh** overnight capacity.

We will be ordering the units from Tesla in Portugal: https://www.tesla.com/pt_PT/powerwall
![image](https://user-images.githubusercontent.com/194400/36089538-5fc8b9b4-0fd4-11e8-98bd-3f1e2f3539a9.png)
(_obvs we are not going to put our battery investment **outside** the building..._)
![image](https://user-images.githubusercontent.com/194400/36089580-8c4a0b5a-0fd4-11e8-88bc-61cef83c2121.png)
![image](https://user-images.githubusercontent.com/194400/36089611-aedd9254-0fd4-11e8-8462-978976e13535.png)
Specs in Portuguese:
![image](https://user-images.githubusercontent.com/194400/36089814-99809644-0fd5-11e8-8fa6-73d520a2adec.png)

# Investment Payback Period?

Assuming that at "full capacity" we will drain the **24kWh** stored in the Powerwall each night,
and a **kWh** price of **€0.1649**: https://www.edp.pt/particulares/energia/tarifarios/
![edp-renewable-kwh-price-0 1649](https://user-images.githubusercontent.com/194400/49269642-e0799380-f45d-11e8-81a4-609eb835ee35.png)

And **200h/month** (_average_) sunshine hours. (200 hours x 12 months / 365 days =) **6.58h/day**
![image](https://user-images.githubusercontent.com/194400/49269491-2bdf7200-f45d-11e8-8e7c-d8ec88baeced.png)
https://weather-and-climate.com/average-monthly-hours-Sunshine,braga,Portugal

On average it **rains** on **157 days** each year. (_note: most of the time it's not **continuous** rain like in England! more like intermittent showers and then the sky clears! Winter is actually relatively sunny!!_)
https://weather-and-climate.com/average-monthly-Rainy-days,braga,Portugal
![image](https://user-images.githubusercontent.com/194400/49269820-e2902200-f45e-11e8-8e27-828e9382d66a.png)
_Conservatively_, we should be able to collect enough energy from our PV cells to power the house during the day and fully charge the batteries for 200 days of the year. 

> **Note**: the sun still shines when it's overcast and solar panels still collect quite a _lot_ of energy. 
But for _simplicity's_ sake, let's _assume_ **200 days** per year.

The "back of the envelope" calculations are: **200 days** x 24 kWh x €0.1649 = **€791.52** per year.
If the Powerwalls cost us **€15k** to install, it will take (15000/791.52 =) **19 Years** to "pay off"!! 😮 
(_this does not factor our "cost of capital" which is currently 2% interest_) 

This is the "worst case" where there is ***zero*** sun for 157 days per year. 
_Obviously_ that does not happen. So a more _realistic_ calculation is using **300 days**:
**300 days** x 24 kWh x €0.1649 = **€1187.28** per year. or (15000/1187.28 =) **13 Years** "pay back".

Obviously these numbers aren't _great_ from a purely _financial_ investment perspective.
Investing the same cash in a portfolio of tech stocks will deliver a _significantly_ higher return!!
So we need to "make the case" for ["being the change we want to see"](https://www.quora.com/Did-Gandhi-really-say-Be-the-change-you-want-to-see-in-the-world) ...
and decide _when_ to make the investment. (_see "Next Action" below_)

 
<hr />

If you want to understand this calculation in more detail, consider watching this video:
**Payback period** for **Tesla PowerWall 2** and 5kW Solar Array: **10 Years** (_conservatively_)
[![solar-payback-time](https://user-images.githubusercontent.com/194400/39960671-8b2d26e0-561e-11e8-8d23-0aa818baba88.png)](https://youtu.be/Sc1rPLhkLQc)
https://youtu.be/Sc1rPLhkLQc

Baring in mind that we will have _considerably_ more people in the house than the "average family"
I feel that 3 x 4KW is the _minimum_ we should go for in terms of Solar Panels.
Note: Electricity Cost in PT is more than _double_ that of Australia (_**11** cents/KWh vs **23** cents/KWh!_)
So the _cost_ of using "Grid" Power in PT is _much_ higher see:
https://www.statista.com/statistics/418111/electricity-prices-for-households-in-portugal
![image](https://user-images.githubusercontent.com/194400/39960935-efb71e04-5623-11e8-98c6-ac4cd4f0d613.png)
The cost of Electricity _might_ come down in PT in the future 
given that "Wholesale" prices have dropped last year (_because of the "renewables glut"_):
https://www.pv-magazine.com/2018/04/12/portugal-renewables-are-driving-down-electricity-prices
![image](https://user-images.githubusercontent.com/194400/39960947-27c81226-5624-11e8-8054-cd94a4f079fb.png)
But I would not "count" on the prices coming down in PT because EDP basically has a Monopoly ...
Further reading: https://smartexpat.com/portugal/how-to-guides/home/services/electricity

# _Next Action_

+ [ ] Determine ***when*** the appropriate time to make this investment decision is ...
  + baring in mind that we don't _need_ to have the batteries installed at "launch" because nobody will book a stay **`@home`** on the basis of us having solar power storage and €15k is quite a lot to pony up before we _need_ 24kWh over night!
  + my **suggestion** is that we re-visit this in **April 2019 _`after`_** we reach "capacity" and have had chance to gauge our energy usage over the winter months.

> **Note**: my _strong_ preference would be to use a Redflow ZCell (Zinc Bromide Flow Batteries) 
> Watch: Fully Charged Redflow ZCell batteries: https://youtu.be/4OHstY_kKUY
> Read: https://www.greentechmedia.com/articles/read/can-redflows-home-flow-battery-really-beat-lithium-ion
> Product Page: https://redflow.com/products/redflow-zcell
> Sadly, at the time of writing **(a)** ZCell **does not ship** to Portugal (_and there are no distributors in the country!_) and **(b)** ZCells are ***more expensive*** than Tesla's Powerwalls. 
> But in ["Phase Two"](https://github.com/dwyl/phase-two) I will be using "Flow" Batteries for _sure_ because Lithium is (_relative_) _scarce_!

> <sup>**1**</sup>Our plan is to combine Solar Panels with Energy Storage (_initially in the form of Tesla Powerwalls, hence this issue_) to capture enough energy during the day to power everything in the house and store enough to power to use at night. Once the Power Walls are fully charged we will supply the surplus solar power back to the Grid so we may end up being a net _exporter_ of power. (TBD!)
We are fully aware that sun does not shine all year, so we'll be supplementing our on-premises Solar with Grid power from EDP Renewables [edpr.com/en](https://www.edpr.com/en) **€0.16/kWh** which comes from Hydro and Wind. Our reasoning for not relying 100% on the grid for power is three-fold: **(a)** demand for electricity is going to sky-rocket in the next 5-10 years with the accelerated adoption of electric vehicles (_thanks Elon, Nissan/Renault, Hyundai, etc!_) so we want to do our part in not straining the grid. **(b)** we are very keen on **Energy Independence** both from a resilience and security perspective. Finally, **(c)** We plan to have an [Electric Smart Car](https://youtu.be/riyLR40nfyM) (_or similar "budget" electric city runabout vehicle, e.g: [Zoe](https://youtu.be/LEIzORzvbdQ) or [Leaf](https://youtu.be/bnIg-aEHD-Y) ..._) for people living in the house to use for short trips. This is a totally separate issue/decision, but it's "on the horizon" and we're planning for it!

> <sup>**2**</sup>@dwyl we pitched to work with/for an Renewal Energy company and in addition to my _natural_ curiosity for renewables I did a "deep dive" into the sector while preparing our proposal and pitch "deck" ... https://github.com/dwyl/hq/issues/417 at the time this could have been my "Mastermind Specialist Subject" https://en.wikipedia.org/wiki/Mastermind_(TV_series)#Specialist_subjects ... 
I still retain much of the info in memory, so feel free to "quiz" me on it if you are curious! 😉

**Comment by nelsonic** [comment]
Asked the HN community for help: https://news.ycombinator.com/item?id=18567051

**Comment by nelsonic** [comment]
Cross posted the question to: https://sustainability.stackexchange.com/questions/6974/is-telsa-powerwall-a-good-investment

**Comment by iteles** [comment]
So, so excited about making the investment in being carbon neutral and showing that it can be done 👍 
Agreed with revisiting in April 2019, but thanks for opening the issue so we can capture research.

**Comment by iteles** [comment]
In light of the amount of energy we will need to power `home` and the fact that the local energy company is dragging their feet on the higher energy provision, I have been reading up on Tesla Powerwalls in more detail as we may well need as many as we can charge ourselves (solar panels to be discussed separately).

## To be added to calculations

### EDP costs
Our new calculations will need to factor in the **costs that the energy company is looking to charge us for the installation of extra equipment for grid electricity**, which we are waiting for.  

### Powerwall installation costs
We will also need to factor in the additional installation costs, which have ***gone up considerably*** since your screenshot two weeks ago 🤦‍♀️  as well as anything EDP charges us to connect to the grid (if this is a thing, need to research this next).
> O custo típico de instalação varia entre € 1.100,00 e € 3.400,00. Isto não inclui a instalação solar, **upgrades elétricos (se necessário), impostos, taxas de permissão ou quaisquer taxas de distribuição/de conexão que sejam aplicáveis**. Esta estimativa inclui um imposto sobre o valor acrescentado (IVA) de 23%.

![image](https://user-images.githubusercontent.com/4185328/50053728-ed6ed600-0131-11e9-9436-a322700496e6.png)

### Accounting for maximum output
Lastly, Powerwalls have a maximum continuous output of 5kW, so even with 3 powerwalls, there will be times when even on a sunny day, with air conditioning, showers and a couple of machines/stoves running, we'll need grid electricity.

### Feed-in tariffs
Feed-in tariffs _could_ be considered but are honestly _negligible_ in Portugal as they are no longer subsidised for solar (since 2015), meaning they are among the lowest in Europe.  
[Source](http://app.parlamento.pt/webutils/docs/doc.pdf?path=6148523063446f764c324679626d56304c334e706447567a4c31684a5355786c5a793944543030764e6b4e4654314176523152425655513257456c4a4c305276593356745a57353062334e4259335270646d6c6b5957526c5132397461584e7a595738765a6d56685a6d59324d4459745a574935596930304d6a466a4c546c6d4d54517459575933595749774d6d4a6c59325a6b4c6e426b5a673d3d&fich=feaff606-eb9b-421c-9f14-af7ab02becfd.pdf&Inline=true) and http://www.res-legal.eu/search-by-country/portugal/single/s/res-e/t/promotion/aid/feed-in-tariff-tarifas-feed-in/lastp/179/ and http://www.res-legal.eu/search-by-country/portugal/tools-list/c/portugal/s/res-e/t/promotion/sum/180/lpid/179/
![image](https://user-images.githubusercontent.com/4185328/50056867-ab0dbf00-015a-11e9-8dff-e698d13b5a76.png)


## 3-phase grid

In the meantime however, there is one thing I've been researching which hasn't yet been mentioned: ***Using Powerwalls with a 3-phase grid***.

`home` will necessitate a 3-phase grid connection. In Portugal, the EDP grid is 3-phase but most small homes only get monophasic or 2-phase connections into the home itself.

So far, the information I have found points to this meaning that the Powerwall 2 can only be connected to a single phase:
http://www.solarmagec.com/home/625-tesla-powerwall-2-.html
![image](https://user-images.githubusercontent.com/4185328/50053560-41c48680-012f-11e9-85b2-0f0fd319a120.png)

Other than having 3 powerwalls, is there a way to overcome the inefficiencies of having one powerwall connected to a single phase, meaning that it will only be able to be used for that single phase rather than all 3 phases?

It looks like there are a few options:
+ **1. Use *one* single-phase hybrid inverter with *three* consumption current transformers**
  + These 3 CTs export enough power during a single phase to then offset the cost from the grid across the other 2 phases once it gets to the 3-phase energy meter
  + Only ***some*** single-phase hybrid inverters allow you to do this: ["Single-phase hybrid inverters controlled by a Reposit controller can do this."](https://www.solarquotes.com.au/blog/home-batteries-3-phase) (Configuration #3) :arrow_left: requires more research into what's available in Portugal
  + Note: `hybrid` inverters are both solar inverters and battery inverters in one - this is likely to be unfeasible with a Powerall because of their built-in inverters (see option 2 below)

![image](https://user-images.githubusercontent.com/4185328/50056260-8eb95480-0151-11e9-9567-2bdc1e62da34.png)

+ **2. Use a *three-phase solar inverter*, one *battery* inverter with *three* consumption current transformers**
  + Same as the option above but splits the solar and battery inverters rather than having a hybrid
  + I believe this would be what we would have to go with for a Powerwall specifically as they have in-built (battery) inverters
+ Having the 3 CTs would mean we could use the battery inverter that's already in the Powerwall with the 3-phase meter from EDP
![image](https://user-images.githubusercontent.com/4185328/50056386-5ca8f200-0153-11e9-931b-cf300c08b8ce.png)

![image](https://user-images.githubusercontent.com/4185328/50057051-1ce70800-015d-11e9-906d-d889177c03ab.png)



+ **3. Use a *three-phase solar inverter*, *three batteries* each with their own battery inverters with *three* consumption current transformers**
  + By far the most expensive option as 3 batteries/powerwalls are involved
  + Each battery is hooked up to a single phase
  + This would only make sense if we have enough solar panels to fill all 3 batteries regularly
![image](https://user-images.githubusercontent.com/4185328/50056957-f2e11600-015b-11e9-8b40-c47c2d206902.png)

+ **4. Use a *micro-inverters* on *each* solar panel, one *battery* inverter with *three* consumption current transformers**
  + This is the same as option 2, but each solar panel has its own micro-inverter - a cursory search show this option to be quite a bit more expensive in Portugal and require more maintenance 
  + There's only one read advantage in that if a single solar panel in a string is shaded, you don't lose the power from the others in that string
  + You also don't have the high DC voltage running through your house to the inverter so there is a bit more safety (though this shouldn't be an issue with good installations)
![image](https://user-images.githubusercontent.com/4185328/50057092-9f6fc780-015d-11e9-9eb1-8c75c2346f19.png)

See here for more on micro-inverters:
[![image](https://user-images.githubusercontent.com/4185328/50057199-ea3e0f00-015e-11e9-8400-5559298e66cb.png)](https://www.solarquotes.com.au/inverters/micro/)



Note: **A 3-phase hybrid inverter** is [_not an option_ for the Tesla Powerwall 2 because it is an AC coupled system](https://www.solarquotes.com.au/blog/home-batteries-3-phase/#config1).

### Additional Questions
+ Can we connect the 5 heat pump solar panels (which will take up a considerable amount of the sun-facing side of the roof) to the batteries too?
+ Is the solar inverter for the heat pumps going to be a 3-phase solar inverter or a single phase (diagram below for illustrative purposes):
![image](https://user-images.githubusercontent.com/4185328/50057020-c24dac00-015c-11e9-95be-8d965641516a.png)

**Comment by nelsonic** [comment]
# ⚠️ New Info: EDP _Saga_ to Increase Power Supply! 😞 

A _big_ part of preparing the house to accommodate more people is increasing power supply from the "domestic" **30 Amp** (6.6kVA) to a _commercial_ **90 Amp** (19.8kVA) power so that multiple people can cook simultaneously and/or use the Air-conditioning without the power cutting out ... ⚡️ 

Sadly, a task we were _assured_ would be "straightforward" when we _started_ this project is turning out to be anything but that! EDP, the company that basically owns a Monopoly on energy _infrastructure_, is now saying that they need to install a "PT" ("Posto de transformação" i.e. high tension transformer box) on our property that will occupy 5m (width) x 3m (length) x 2.5 (height) ... 
these are the examples we've been given:  [postos-de-transformação](https://www.eem.pt/pt/conteudo/sistema-el%C3%A9trico/distribui%C3%A7%C3%A3o/postos-de-transforma%C3%A7%C3%A3o/)
![image](https://user-images.githubusercontent.com/194400/50058130-04cab500-016c-11e9-8256-e7b0581a5fef.png)

The "technician" from EDP claims that it's not a matter of giving us 3 x 30 Amp cables from the _existing_ transformer to our property, that we _must_ have this hideous eyesore on the property if we are to increase the electrical capacity of the house ... 

## How Much Power do We _Need_...?

We calculated that our **`PEAK`** power consumption could be well in excess of the 30A the house currently receives from EDP; in fact we could _easily_ exceed 100 Amps just by having the Air Conditioning units on simultaneously and using the Oven and Hob to make food ...

Our Google doc spreadsheet for calculating **`PEAK`** power consumption is:
https://docs.google.com/spreadsheets/d/1_87JQzwjoNmvGEIQ9PJwo3jtNqoDnAvqQTfTchN3UuQ
![image](https://user-images.githubusercontent.com/194400/50060228-3605ae00-0189-11e9-9574-f58d77a74c61.png)

Our _baseline_ consumption will be around **6.725 kWh** including all the _essentials_ such as Lights, Water heating, Laptops, Fridges/Freezers and Security systems. 
This is the "_good news_" ... i.e. we _can_ open the house to a _handful_ of people _without_ increasing the power in the house. _However_ if we switch on _any_ other high-energy appliance 
(_e.g: a Kettle, Oven or Heater_) it will overload the system and cause a blackout in the house.

We _need_ to increase the power in order to be able to use the "nice things" e.g AirCon/Heating, Dish washers, tumble driers, heated pool, spa and sauna! 😮 

### 30 Amp x 220 Volts = 6.6 kW

Using [Ohm's law](https://en.wikipedia.org/wiki/Ohm's_law) we have a _simple_ formula for calculating Power and/or Current: 
If the stated Amperage (I) of the house is **30A** (_which is perfectly acceptable for a single family dwelling_),
And the voltage is 220V @ 50 Hz in Portugal.
We can multiply the I x V to get Watts ... **6.6 kW** in our case.

https://www.rapidtables.com/calc/electric/Amp_to_Watt_Calculator.html
![image](https://user-images.githubusercontent.com/194400/50052724-79780200-0120-11e9-997c-08a01490a8df.png)

If your high school physics is a bit rusty, see: https://www.wikihow.com/Calculate-Wattage

## How to increase electrical capacity?

The EDP website makes it _look/sound_ like it's relatively easy to increase the electrical capacity:
https://www.edp.pt/particulares/apoio-cliente/perguntas-frequentes/novo-contrato/o-que-preciso-de-saber-para-contratar/o-que-e-a-potencia-contratada-e-qual-devo-escolher/faq-4676
![image](https://user-images.githubusercontent.com/194400/50060407-76fec200-018b-11e9-8823-692779d32bd0.png)


## Bi-horária ("Off-peak" Tariff) €0.0974 / kWh

![image](https://user-images.githubusercontent.com/194400/50042790-4ff7a180-0060-11e9-90f4-56ca253ba3c0.png)

https://www.edp.pt/particulares/apoio-cliente/perguntas-frequentes/tarifarios/o-que-preciso-de-saber-para-contratar/o-que-e-a-opcao-horaria-e-qual-a-melhor-para-mim/faq-4823

![image](https://user-images.githubusercontent.com/194400/50041681-64cc3900-0050-11e9-8244-26d350be7063.png)

Probably the _easiest/best_ way of viewing EPD's pricing table is _not_ on the EDP website,
but rather on the "Light Shop" comparison website: 
https://lojaluz.com/fornecedores/edp/tarifas/preco-kwh
![image](https://user-images.githubusercontent.com/194400/50082635-58c7af00-01e9-11e9-8896-975ca725bef4.png)

# €0.0731 / kWh Off-peak




### Can we use _Cheaper_ Off-peak Power to Charge the Powerwall? €0.0731/kWh

If you need a succinct _video_ explaining this, see: Charging the Tesla PowerWall 2 from grid:
[![image](https://user-images.githubusercontent.com/194400/50042108-3bfb7200-0057-11e9-9529-305dd144f73b.png)](https://youtu.be/xCKQ1LQCyX0)
https://youtu.be/xCKQ1LQCyX0

![image](https://user-images.githubusercontent.com/194400/50042136-c17f2200-0057-11e9-9d5b-6835485ef86d.png)


<hr />

### The Future: Increasing Power Consumption 

It's virtually _guaranteed_ that (national/aggregate) power _consumption_ is going to continue rising thanks to the adoption of **electric vehicles** for _personal_, public transport and commercial use.
There are electric-only parking spaces in _most_ shopping malls in Portugal and a charging station 100m from the Braga house; mainstream adoption is coming. 

> Look no further than the most recent Geneva Motor Show where there were more Electric vehicles than "ICE" https://www.wired.co.uk/article/electric-cars-geneva-motor-show-2018
e.g: Jaguar i-Pace, Hyundai Ioniq & Kona/Kauai, VW E-Golf, [Merceds EQC](https://www.mercedes-benz.ca/en/eq-electric-cars), Audi e-Tron, Porsche Mission-E, Mini Cooper Electric, Honda Clarity, Nissan Leaf 2, Mahindra KUV100, BYD e6, etc.
watch Fully Charged Geneva 2018 coverage: [youtu.be/HfWOTMz5HUI](https://youtu.be/HfWOTMz5HUI)

The increased national consumption will _inevitably_ put pressure on supply
meaning that power _failure_ is _likely_ even if (_regardless of whether_) we increase our supply amperage.
Additional signs that power consumption trend will continue:
+ https://en.wikipedia.org/wiki/Tesla_Gigafactory_Europe
+ https://futurism.com/elon-musks-former-employee-is-building-a-european-rival-to-teslas-gigafactory

https://en.wikipedia.org/wiki/World_energy_consumption
![image](https://user-images.githubusercontent.com/194400/50079720-4007cb00-01e2-11e9-84b6-eb4caf200fb6.png)


<hr />

## Power Cuts?

Electrical power is something _most_ people who live in "_developed_" countries take for grated ...
Many _energy rich_ countries use **ten times** more power/electricity per person than in _developing_ nations. 
https://en.wikipedia.org/wiki/List_of_countries_by_electricity_consumption (_sort by average kWh/capita_)
Note: basically _ignore_ the Nordic countries who have extremely cold winters 
and use their abundant Geothermal energy to heat their homes ... 
![top-energy-consumers](https://user-images.githubusercontent.com/194400/50047980-8c56ec00-00b8-11e9-991e-6a83e8510f90.png)
Notice how average energy consumption (per person) in the USA is **12 _Million_ Watt Hours** (12,071 kWh)
Whereas in the **UK 4,795 kWh** and **Portugal 4,245 kWh** per person per year respectively. 
That's almost 3 times as much power use. This might be due to in part to _norther_ US states skewing the numbers because, like the Nordic countries (_which lead the power consumption table by a wide margin_) heating homes in the freezing winter takes a _lot_ of energy! But it also has a lot to do with the southern states using Air Conditioning 24/7 in the summer which uses _obscene_ amounts of power! 💸 
![uk-and-portugal](https://user-images.githubusercontent.com/194400/50047986-ae506e80-00b8-11e9-8893-ad8d030b8974.png)
People in countries at the ["bottom of the pyramid"](https://en.wikipedia.org/wiki/Bottom_of_the_pyramid) essentially have almost no _access_ to electricity,
in many cases they use diesel generators which are not counted as "grid" generated electricity.
![bottom-gaza-strip](https://user-images.githubusercontent.com/194400/50047998-e5bf1b00-00b8-11e9-87ff-4d0ed4c7e50c.png)

The _fact_ is that for _most_ humans around the world power cuts are still a _regular_ occurrence in 2018
https://www.worldatlas.com/articles/power-outages-in-places-of-business-around-the-world.html
![image](https://user-images.githubusercontent.com/194400/50047785-5c0d4e80-00b4-11e9-9c14-17c8c50b925f.png)

***Portugal*** has a **reasonable _reliably_** electrical supply (_due mostly to the sustained investment in [Hydro](https://en.wikipedia.org/wiki/List_of_hydroelectric_power_stations_in_Portugal) during the [Salazar](https://en.wikipedia.org/wiki/Ant%C3%B3nio_de_Oliveira_Salazar) years and massive investment in [Wind](https://github.com/dwyl/home/issues/20) over the last decade_) 
Sadly, power supply is still not 100% guaranteed. see: http://portugalresident.com/edp-cuts-all-the-time
And in 2017: http://theportugalnews.com/news/lisbon-metro-in-the-dark-following-power-cut/42162

The average is **2 minutes per month** of power cuts. Not an issue if all you're doing is watching TV,
but if your house has critical security systems that require power such as Doors and servers/NAS which will suffer if power is "yanked", we basically need to re-think the priority of Tesla Powerwall ... 💭 


# Use Powerwall as _Backup_ Power Supply?

Watch this video: 5 Awesome (and Not-so-awesome) things about the Tesla PowerWall 2
[![image](https://user-images.githubusercontent.com/194400/50042326-7f56e000-0059-11e9-95f3-6b10c988cb87.png)](https://youtu.be/5wFXCrfM140)
https://youtu.be/5wFXCrfM140

![image](https://user-images.githubusercontent.com/194400/50042231-2edf8280-0059-11e9-873c-a0c6a6b57bce.png)

![image](https://user-images.githubusercontent.com/194400/50082504-130ae680-01e9-11e9-9c2f-4a0aa0992e42.png)

> Must remember to factor 30% degradation of battery efficiency into calculations. (see below).


## Tesla Powerwall 2 Local _Gateway API_

The 

According to this GitHub repo: https://github.com/vloschiavo/powerwall2 
The Tesla Powerwall 2 runs a Linux server with a _read-only_ web UI 
that any web browser can connect to on the home/internal network:

![image](https://user-images.githubusercontent.com/194400/50043817-b4236100-0072-11e9-8747-db7a89291703.png)

This means that:
**a)** _Anyone_ in the house can view the current energy consumption in the house without needing to download the Tesla App. Therefore _everyone_ can be a ["***conscious consumer***"](https://medium.com/we-learn-we-grow/lets-define-conscious-living-for-the-21st-century-336fcba332dc) and avoid _wasting_ energy. 
**b)** We can display the _current_ energy consumption (_via both the @home App and on strategically located e-ink displays_) for each room in the house in real-time. Play a warning sound/alarm on each connected device when we are approaching the maximum power consumption for the house. 
That way _everyone_ can know when we are reaching our max load point and turn off that "extra" induction hob or AirCon unit!

## Notes

+ Solar panels still work 10-20% on cloudy days:
https://cleantechnica.com/2018/02/08/solar-panels-work-cloudy-days-just-less-effectively/
+ Electrical Load Management: http://large.stanford.edu/courses/2010/ph240/malone1
read this to understand load shifting.
+ Tesla installation process:
https://www.tesla.com/support/energy/install/powerwall/powerwall-installations
+ Powerwall JSON API:
https://teslamotorsclub.com/tmc/threads/powerwall-2-gateway-api-documentation.112341/
+ Induction hob power consumption:
https://forums.moneysavingexpert.com/showthread.php?t=5652536

## Micro Solar Generation Feed in Tariff (FIT) €0.095 / kWh

If we install 8kWh of Solar power and have fully charged Powerwall we can take advantage of the FIT to turn back our electricity meter. If we **_buy_ off-peak** at **€0.0731/kWh** and _use_ it during the peak time we could use the Powerwall for "load shifting". 
We could even _sell_ during peak time as "arbitrage", but this is unlikely to be profitable as it won't offset the cost of the Powerwall depreciation. 

https://www.pv-magazine.com/2018/01/24/portugal-announces-fit-distributed-solar-generation-fit-of-e95-mwh-for-2018/

# _Next Action_

+ [ ] Calculate payback period for 2 or 3 Powerwall units. 
  + [ ] Charging from the grid at night on cloudy/rainy days at €0.0731/kWh
  + [ ] Useful life of batteries 10 years.  
(_even if they last longer, we should 100% depreciate by the end of the "useful life"_)
  + [ ] Factor in **_30% battery efficiency degradation_** over 10 years. 
  + [ ] Factor in "_round trip efficiency_" of **_87%_** ... see: 
    + https://en.wikipedia.org/wiki/Tesla_Powerwall#Technology 
    + "_**real-world AC-AC round trip efficiency** drops to **87%**_" 
    http://www.catalyticengineering.com/top-ten-facts-about-teslas-350kwh-powerwall-battery/
+ [ ] Contact Tesla Portugal and ask them for the following info:
  + [ ] Price and timelines for installing _Two_ Powerwall units with grid charging/backup functionality.

**Comment by nelsonic** [comment]
Powerwall Long-term review: https://youtu.be/6s6kN9Ezws0 (18 Dac 2018)

**Comment by nelsonic** [comment]
https://www.tesla.com/findus/list/stores/Portugal

El Corte Inglés Av. António Augusto de Aguiar 31
1069-413 Lisbon

@iteles do we have time to go past Tesla on our way home today?

**Comment by iteles** [comment]
@nelsonic Only if we delay our fire safety meeting to tomorrow (perfectly doable) as otherwise by the time we get to Braga it will be dark (sunset at 17h30).

**Comment by nelsonic** [comment]
@iteles I feel that getting a "face-to-face" meeting at Tesla (_even if it's just with a "sales rep"_) 
is worth the effort. To be clear: El Corte Inglés is pretty much only a 30 mins "detour" from our path.
So we _could_ still make it to Braga by 17:00 if we leave Lx no later than 13:30 
https://goo.gl/maps/9TNwoBvQ6LU2
![image](https://user-images.githubusercontent.com/194400/51167473-99751f80-189e-11e9-95ee-47831c31d515.png)

What time have you agreed with the fire safety people? (_separate issue but related to today..._)

**Comment by iteles** [comment]
The proposal we have been given to include batteries 

## For 2 sections of the roof
![image](https://user-images.githubusercontent.com/4185328/58373253-e5a62f80-7f22-11e9-99cc-d4d1e72816b7.png)

24  x 315W panels and **9kW of storage** via two 4.5kW batteries.

We have been quoted **€ 7 182,00** (plus VAT) for the setup **without batteries** and €11 944,80 (plus VAT) with batteries** (this includes a different inverter and the batteries themselves).

Battery setup = €4, 762.80 more expensive or **€5,858.24** including VAT.

## For 3 sections of roof (if this is even possible)
![image](https://user-images.githubusercontent.com/4185328/58373279-88f74480-7f23-11e9-9e22-6e268b29b2e2.png)

32  x 315W panels and **12.6kW of storage** via two 6.3kW batteries.

We have been quoted **€ 10 176,00** (plus VAT) for the setup **without batteries** and €16 981,00 (plus VAT) with batteries** (this includes a different inverter and the batteries themselves).

Battery setup = €6,805 more expensive or **€8,370.15** including VAT.


## Spec for the proposed batteries
![IMG-20190423-WA0002](https://user-images.githubusercontent.com/4185328/58373336-9c56df80-7f24-11e9-9627-3d6d50d37ef5.jpg)
[Triple-Power_cn-batteries-by-SolaX.pdf](https://github.com/ideaq/home/files/3219614/Triple-Power_cn-batteries-by-SolaX.pdf)

**Comment by iteles** [comment]
@nelsonic I _really_ want to invest in these batteries as an MVP to test out the ability of being energy-independent for a portion of the year.

According to the calculations you laid out in the original post here:
200 days x 12.6 kWh x €0.1649 = **€415.548**, which means it would take us **20.14 years to pay off the batteries** (far longer than their warranty). 😞 

From a _purely financial_ perspective, this doesn't make sense.

However, I feel that if we want to make a true effort to be energy conscious and 'green', this is the time. If we want to extend this beyond 'home' in the future, having hard data from this place is the _only_ way to convince anyone who may want to join us in the future that this is a worthwhile endeavour. Of course I am under no illusion that people outside of the two of us will appreciate the investment for now!

Secondly, the batteries provide us with a failsafe. Should there ever be a power outage, we can still run basic security systems from the batteries.

**Comment by iteles** [comment]
@nelsonic Given this is a big investment I wanted to check: Is this thumbs up an acknowledgement of my thoughts on this or an agreement we should give the go-ahead for the batteries?

![image](https://user-images.githubusercontent.com/4185328/58761402-f450a000-853b-11e9-93f8-acf4239962e1.png)

**Comment by nelsonic** [comment]
@iteles we have _already_ agreed that the Batteries are a security feature.
The EDP "box" with the main power cutoff is on the street outside the house. 
Anyone with a 3D-printed plastic key can get into the EDP box and cut the power to the house.
by doing this they disable all our security systems and can access the house unimpeded. 
I'm sorry I did not write a more _conclusive_ reply when I checked my notifications on my phone this morning. But yes, as per our original budget, we are getting the batteries.
